components/log/log_noos.c

```diff
@@ -7,18 +7,10 @@
 #include <assert.h>
 #include "esp_log_private.h"
 
-#ifdef CONFIG_IDF_TARGET_ESP32C3
-#include "hal/cpu_hal.h"  // for cpu_hal_get_cycle_count()
 #ifdef __ZEPHYR__
 #include <zephyr/kernel.h>
 #endif
-#else 
-#ifdef __ZEPHYR__
-#include <zephyr/kernel.h>
-#endif
-#include "hal/cpu_hal.h"  // for cpu_hal_get_cycle_count()
-#endif
-
+#include "hal/cpu_hal.h"
```

Comment:
Have you tested this specific change (cpu_hal.h) placement for all SoCs? AFAIK, C3 required a different include sorting, which caused build warnings.

Reply:
I've tested on all SOCs and all builds without warnings, so I did this simplification in includes.